generator/internal/parser/protobuf.go

```diff
@@ -163,14 +173,34 @@ func determineInputFiles(source string, options map[string]string) ([]string, er
 			return nil
 		}
 		if filepath.Ext(path) == ".proto" {
-			files = append(files, path)
+			files[path] = true
 		}
 		return nil
 	})
-	if err != nil {
-		return nil, err
+}
+
+func applyIncludeList(files map[string]bool, sourceDirectory string, options map[string]string) {
+	list, ok := options["include-list"]
+	if !ok {
+		return
+	}
+	// Ignore any discovered paths, only the paths from the include list apply.
+	for k := range files {
+		files[k] = false
+	}
+	for _, p := range strings.Split(list, ",") {
+		files[path.Join(sourceDirectory, p)] = true
```

Comment:
We allow specifying additional roots. Because of this I wonder if these fields should be absolute paths. Thoughts?

Reply:
I believe `sourceDirectory` is always interpreted as "relative to *one* of the roots", we find the directory in one of the roots, and then find the files (or apply the include list / exclude list) to the files found in that directory.